install.py

```diff
@@ -477,8 +533,21 @@ def validate_path(path):
         cmake_flags += [f"-Dlegate_core_LEGION_BRANCH={legion_branch}"]
     if build_docs:
         cmake_flags += ["-Dlegate_core_BUILD_DOCS=ON"]
+    if build_tests:
+        cmake_flags += ["-Dlegate_core_BUILD_TESTS=ON"]
     if build_examples:
-        cmake_flags += ["-Dlegate_core_EXAMPLE_BUILD_TESTS=ON"]
+        cmake_flags += ["-Dlegate_core_BUILD_EXAMPLES=ON"]
+
+    if not is_conda and editable:
```

Comment:
I just did what seemed to work in a conda env and pip venv, but I agree the logic is super weird.

Note: `CMAKE_INSTALL_PREFIX=<skbuild_dir>/cmake-install` for all the commands below.

### In a pip virtual env

We install `_deps/legion-build/bindings/python` into `${CMAKE_INSTALL_PREFIX}` (i.e. `<skbuild_dir>/cmake-install`).

This produces the following cmake install rule for Legion's python bindings:
```shell
${VIRTUAL_ENV}/bin/python3 \
    -m pip install \
    --global-option=--cmake-build-dir "--global-option=<skbuild_dir>/cmake-build/_deps/legion-build/runtime" \
    --global-option=--prefix "--global-option=${CMAKE_INSTALL_PREFIX}" \
    --ignore-installed \
    --root / \
    --prefix ${CMAKE_INSTALL_PREFIX} .
```

Because there's no `--editable` flag, pip installs the Legion Python bindings into `${VIRTUAL_ENV}/<..>/site-packages`.

During this process, the Legion Python bindings' `setup.py` inlines the `--prefix` arg into the generated `legion_canonical_cffi.py`:
```python
libpath = "<skbuild_dir>/cmake-install/lib/legion_canonical_python.so"
```

Alternatively, if we installed into `${VIRTUAL_ENV}`, the generated `legion_canonical_cffi.py` would have this path:
```python
libpath = "${VIRTUAL_ENV}/lib/legion_canonical_python.so"
```

This leads to dlopen'ing a non-existent file at runtime, because the Legion Python bindings _do not_ install `liblegion_canonical_python.so` into `${VIRTUAL_ENV}/lib` -- for that we'd have to install all of Legion into `${VIRTUAL_ENV}`, defeating the purpose of editable installs.

### In a conda env

Install `_deps/legion-build/bindings/python` into `${CONDA_PREFIX}`, yielding the following CMake install rule:
```shell
${CONDA_PREFIX}/bin/python3 \
    -m pip install \
    --global-option=--cmake-build-dir "--global-option=<skbuild_dir>/cmake-build/_deps/legion-build/runtime" \
    --global-option=--prefix "--global-option=${CMAKE_INSTALL_PREFIX}" \
    --ignore-installed \
    --root / \
    --prefix ${CONDA_PREFIX} .
```

The above still installs the Legion Python bindings into `${CONDA_PREFIX}/<..>/site-packages`, and generates the following `legion_canonical_cffi.py`:
```python
libpath = "<skbuild_dir>/cmake-install/lib/legion_canonical_python.so"
```

This is the correct libpath when doing an `--editable` install in a conda env, but the exact opposite behavior from pip.

### Discrepencies

Let's revisit the Legion Python bindings' install rules for each flavor of venv:
```shell
# pip venv (not working):
python -m pip install ... --global-option=--prefix "--global-option=${CMAKE_INSTALL_PREFIX}" --prefix ${VIRTUAL_ENV} .
# pip venv (working):
python -m pip install ... --global-option=--prefix "--global-option=${CMAKE_INSTALL_PREFIX}" --prefix ${CMAKE_INSTALL_PREFIX} .
# conda env (working):
python -m pip install ... --global-option=--prefix "--global-option=${CMAKE_INSTALL_PREFIX}" --prefix ${CONDA_PREFIX} .
```

All of the above commands pass `--prefix` as a special "--global-option" thing, which (IIUC) is supposed to be forwarded on to `setup.py` and not handled by pip/setuptools. This should mean that the [parsed `--prefix` in Legion's setup.py](https://gitlab.com/StanfordLegion/legion/-/blob/master/bindings/python/setup.py?ref_type=heads#L14-15) should be the same. Except it's not.

* When installing into a pip venv, the `--prefix` `setup.py` sees is the bare `--prefix`
* When installing in a conda env, the `--prefix` `setup.py` sees is the wrapped `--global-option=--prefix`

:exploding_head: :exploding_head: :exploding_head: :exploding_head: :exploding_head: :exploding_head: :exploding_head: :exploding_head: :exploding_head: :exploding_head: :exploding_head: :exploding_head: 

### In conclusion

This wouldn't be as painful if Legion's setup.py was using a proper PEP517 build backend, could supported `--editable` installs, etc. But as is, we're at the whims of pip and setuptools authors, and the builds are brittle and confusing.